torch/_inductor/ir.py

```diff
@@ -7515,6 +7519,17 @@ def create_output(output: IRNode, ind: int):
                     skip_size_stride_alignment_checks=True,
                 )
 
+        # Force the output strides to be same as the original strides
+        new_outputs = []
+        fake_outputs = V.graph.current_node.meta["val"]
+        for idx, output in enumerate(outputs):
+            if isinstance(output, (ShapeAsConstantBuffer, NoneAsConstantBuffer)):
+                new_outputs.append(output)
+            else:
+                example_stride = handle_sym_expr(fake_outputs[idx].stride())
+                new_outputs.append(cls.require_exact_strides(output, example_stride))
```

Comment:
I'm not sure this is right. Can Inductor passes change the fake_outputs in a way that they differ from eager?

If so we need to record the meta vals at the time of tracing, before passes run, and then use the metadata on them.

Reply:
I guess this applies to inputs of the invoke subgraph then as well. Currently, we rely on the meta vals of the inputs of invoke subgraph, which could be different from eager because of graph passes

Reply:
Can you remind me why we want to force the inputs and output strides to be the same as eager? If we were not doing invoke_subgraph, inductor is allowed to change intermediates in the graph to have whatever strides it wants, with some exceptions.

Reply:
This is to reduce compile time. We compile a subgraph once and then call the same subgraph output code on second call. Since the input strides can be different for different subgraph calls, we restride the input to a fixed value at the beginning of each subgraph. 

This allows us to reuse the output code of a subgraph. This is very important for compile time, otherwise the major benefits of invoke subgraph are not realized.

Reply:
We have some infrastructure to do this already (for inputs), check out https://github.com/pytorch/pytorch/blob/bc11afd41fc68637000dcd9c14128c75d9361443/torch/fx/experimental/proxy_tensor.py#L1127-L1134

Reply:
Yea - let's use the above mechanism

Reply:
I can use this for `input`. Is there anything for the `output` strides?